File: DSA/done/1422 - maxScore.py
# https://leetcode.com/problems/maximum-score-after-splitting-a-string/

class Solution:
    def maxScore(self, s: str) -> int:
        total_zeroes = s.count('0')
        cur_zero = ans = 0
        for i in range(1, len(s)):
            if s[i - 1] == '0':
                cur_zero += 1
                total_zeroes -= 1
            ans = max(ans, cur_zero + len(s) - total_zeroes - i)
        return ans

File: DSA/done/1422 - maxScore.java
// https://leetcode.com/problems/maximum-score-after-splitting-a-string/

class Solution {
    public int maxScore(String s) {
        int totalZeroes = 0;
        for (char c : s.toCharArray()) {
            if (c == '0') totalZeroes++; 
        }

        int curZero = 0, maxScore = 0;
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i - 1) == '0') {
                curZero++;
                totalZeroes--;
            }
            maxScore = Math.max(maxScore, curZero + (s.length() - i - totalZeroes));
        }

        return maxScore;
    }
}
